### DISCUSSION TITLE
How overwite etc/nginx/nginx.conf or edit it?

### DISCUSSION THREAD

**Discussion by Sir-Doyle** [NginxProxyManager/nginx-proxy-manager#2073] [Q&A]
Hi all,

Please any idea how can I make a custom configuration of the main of Nginx (etc/nginx/nginx.conf) or if there is any way to edit it?

Thanks

**Comment by jzcad1828** [comment]
This should work, assuming a rootful docker deployment:

Create a local folder to store the nginx.conf file:
`mkdir /opt/npm-nginx-conf`

Copy nginx.conf out of running NPM container to local folder:
`docker cp <container-name>:/etc/nginx/nginx.conf /opt/npm-nginx-conf/nginx.conf`

Recreate NPM container using the original options with a new volume mount:

If using docker run or create, add the following volume directive:
`-v /opt/npm-nginx-conf/nginx.conf:/etc/nginx/nginx.conf:ro #(use :ro,z if enforcing SELinux)`

If using docker-compose, add the following to the volumes section of the container:
```
volumes:
  - /opt/npm-nginx-conf/nginx.conf:/etc/nginx/nginx.conf:ro #(use :ro,z if enforcing SELinux)
```

You should now have a persistent and customizable nginx.conf file for the container. Restart the container after making changes to the conf to apply them.

***NOTE: I have not tested this process with NPM, and this is probably not supported, use at your own risk! :P
